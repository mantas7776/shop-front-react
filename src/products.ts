export const products = [
    {
        id: 1,
        name: "Jacobs Kronung, 250g",
        price: 2.99,
        image: "https://www.barbora.lt/api/Images/GetInventoryImage?id=b5b83d4c-90f6-49da-89e7-138f3382ebab",
        description: "Jacobs Kronung kava, pagaminta iš kruopščiai atrinktų ir rūpestingai paskrudintų kavos pupelių."
    },
    {
        id: 2,
        name: "Seedless green grapes, 1 kg",
        price: 2.29,
        image: "https://www.barbora.lt/api/Images/GetInventoryImage?id=fe89254c-ec64-46f2-b2e2-17bc5e5c5874",
        description: "Žaliosios besėklės vynuogės THOMPSON(2 kl.)"
    },
    {
        id: 3,
        name: "Crisps Estrella, 140g",
        price: 1.38,
        image: "https://www.barbora.lt/api/Images/GetInventoryImage?id=eea3134e-4b0c-419e-bbad-6232f84e1fa5",
        description: "Bulvės, saulėgrąžų aliejus, druska, IŠRŪGŲ milteliai (iš PIENO) (1,7%), svogūnų milteliai (1%), cukrus, PASUKŲ milteliai (iš PIENO) (0,5%), kvapioji medžiaga, mielių ekstraktas, mielių milteliai, česnakų milteliai, prieskoninės žolelės, rūgštis: PIENO rūgštis."
    },
    {
        id: 4,
        name: "Crisps Taffel San Diego, 120g",
        price: 1.28,
        image: "https://www.barbora.lt/api/Images/GetInventoryImage?id=39ca829f-6f9f-4409-a089-711ce6f46049",
        description: "Pomidorų skonio bulvių traškučiai TAFFEL SAN DIEGO."
    },
    {
        id: 5,
        name: "Doritos, 100g",
        price: 0.98,
        image: "https://www.barbora.lt/api/Images/GetInventoryImage?id=87d60690-7504-4a30-9a53-7c3892a7c743",
        description: "Sūrio skonio kukurūzų traškučiai Doritos."
    },
    {
        id: 6,
        name: "Chocolate Milka Choco Biscuit, 300 g",
        price: 3.50,
        image: "https://i.pinimg.com/originals/9d/80/d3/9d80d3dcd95994dfc7a069632476cbf2.png",
        description: "Cukrus, alyvpalmių aliejus, kakavos sviestas, nugriebto PIENO milteliai, išrūgų milteliai (iš PIENO), trinta kakava, KVIETINIAI miltai, PIENO riebalai, emulsikliai (SOJŲ lecitinas, saulėgrąžų lecitinas), LAZDYNŲ RIEŠUTŲ masė, tešlos kildymo medžiagos (E503, E500, E341), dekstrozė, gliukozės-fruktozės sirupas, druska, kvapiosios medžiagos."
    },
    {
        id: 7,
        name: "Chocolate Cookies Milka Cake & Choc, 120 g",
        price: 1.56,
        image: "http://fmcgimport.com/media/catalog/product/cache/1/small_image/210x/9df78eab33525d08d6e5fb8d27136e95/m/i/milka_cake_choc.png",
        description: "SAUSAINIAI SU PIENINIU ŠOKOLADU (28%) IŠ ALPIŲ PIENO."
    },
    {
        id: 8,
        name: "Sausučiai Javinė su aguonomis, 120 g",
        price: 0.54,
        image: "https://www.barbora.lt/api/Images/GetInventoryImage?id=6aa7d94b-1356-4240-abae-a2580028e72c",
        description: "Sausučiai Javinė 120g pakuotė"
    },
    {
        id: 9,
        name: "Šokolado skonio tortas su citrininiu morenginiu traškučiu, 1kg",
        price: 11.99,
        image: "https://www.barbora.lt/api/Images/GetInventoryImage?id=561bb452-80ba-4ebd-9e4b-087488f84a36",
        description: "Sudėtyje yra: glitimo turinčių javų, pieno ir jo produktų (įskaitant laktozę), riešutų ir jų produktų, sojų pupelių ir jų produktų, kiaušinių ir jų produktų"
    },
    {
        id: 10,
        name: "Jack Daniel's Whiskey, 3l",
        price: 89.99,
        image: "https://www.weinquelle.com/fotos/s3779.png",
        description: "Kvietiniai traškučiai, pagardinti aguonomis."
    },
    {
        id: 11,
        name: "Jaegermeister, 0,7l",
        price: 15.99,
        image: "http://www.mast-jaegermeister.co.uk/wp-content/uploads/2015/11/bottle-left.png",
        description: "Charakteringo aromato saldžiarūgštis gėrimas gaminamas iš 27 žolelių antpilo."
    },

]