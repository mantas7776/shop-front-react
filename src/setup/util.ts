import Toster from 'toastr';
import $ from 'jquery'

Toster.options = {
	closeButton: true,
	timeOut: 8000,
	extendedTimeOut: 30000,
	showMethod: 'slideDown',
	hideMethod: 'slideUp',
	hideDuration: 300,
	progressBar: true,
	onShown: function () {
		var toasts = $('#toast-container .toast');
		if (toasts.length > 3) toasts.slice(3).click();
	}
};

class TosterMessenger {
	succ = (message: string) => Toster.success(message);
	err = (message: string) => Toster.error(message);
	info = (message: string) => Toster.info(message);
	warn = (message: string) => Toster.warn(message);
}

export const toster = new TosterMessenger();