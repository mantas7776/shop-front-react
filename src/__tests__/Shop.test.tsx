import React from 'react';
import Shop from '../components/shop/shop';
import products from './fixtures/products.json';
import { Provider } from 'mobx-react';
import Cart from '../stores/cart';
import { shallow, mount } from 'enzyme';
import { toJS, values } from 'mobx';
import ProductComponent from '../components/shop/product';
import '../setup/setup.tsx';

it('Should match snapshot', () => {
	const shopPage = shallow(<Shop products={products} />);
	console.dir(shopPage);
	expect(shopPage).toMatchSnapshot();
});

it('Should render product properly', () => {
	const cart = new Cart();
	const productElement = shallow(
		<Provider cart={cart}>
			<ProductComponent product={products[0]} />
		</Provider>
	);
	expect(productElement).toMatchSnapshot();
});

it('Should add to cart successfully', () => {
	const cart = new Cart();
	const productElement = mount(
		<Provider cart={cart}>
			<ProductComponent product={products[0]} />
		</Provider>
	);
	productElement.find('input').simulate('change', { target: { value: 2 } });
	productElement.find('button').simulate('click');

	expect(cart.cart[0].amount).toBe(2);
});

it('Should add product to cart', () => {
	const cart = new Cart();
	expect(cart.cart.length).toBe(0);

	cart.add(products[0], 1);
	const cartObj = toJS(cart.cart[0]);
	expect(cartObj).toEqual({ ...products[0], amount: 1 });
	expect(cartObj.amount).toBe(1);
	expect(cart.cart.length).toBe(1);
});

it('Should clean cart', () => {
	const cart = new Cart();
	expect(cart.cart.length).toBe(0);

	cart.add(products[0], 1);
	expect(cart.cart.length).toBe(1);

	cart.cleanCart();
	expect(cart.cart.length).toBe(0);
});

it('Should add same product to cart', () => {
	const cart = new Cart();
	expect(cart.cart.length).toBe(0);

	cart.add(products[0], 1);
	cart.add(products[0], 1);
	const cartObj = toJS(cart.cart[0]);
	expect(cartObj).toEqual({ ...products[0], amount: 2 });
	expect(cartObj.amount).toBe(2);
	expect(cart.cart.length).toBe(1);
});

it('Should remove product from cart', () => {
	const cart = new Cart();
	cart.add(products[0], 1);

	cart.remove(products[0]);
	expect(cart.cart.length).toBe(0);
});

it('Should remove nothing from cart', () => {
	const cart = new Cart();

	expect(cart.cart.length).toBe(0);
	cart.remove(products[0]);
	expect(cart.cart.length).toBe(0);
});

it('Should correctly calculate total', () => {
	const cart = new Cart();
	const amount = 3;
	let product = products[0];
	let total = product.price * 3;
	cart.add(product, 3);
	expect(cart.total).toBe(total);
});
