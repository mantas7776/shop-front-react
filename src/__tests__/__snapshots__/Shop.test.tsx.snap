// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should match snapshot 1`] = `
<div
  className="container"
>
  <div
    id="product-list"
  >
    <inject-ProductComponent-with-cart
      key="1"
      product={
        Object {
          "description": "test",
          "id": 1,
          "image": "https://www.barbora.lt/api/Images/GetInventoryImage?id=0198db1a-2a5e-47df-94b0-562be3c7fa8b",
          "name": "test2",
          "price": 10,
        }
      }
    />
    <inject-ProductComponent-with-cart
      key="2"
      product={
        Object {
          "description": "test2",
          "id": 2,
          "image": "https://www.barbora.lt/api/Images/GetInventoryImage?id=0198db1a-2a5e-47df-94b0-562be3c7fa8b",
          "name": "test4",
          "price": 15,
        }
      }
    />
    <inject-ProductComponent-with-cart
      key="3"
      product={
        Object {
          "description": "test3",
          "id": 3,
          "image": "https://www.barbora.lt/api/Images/GetInventoryImage?id=0198db1a-2a5e-47df-94b0-562be3c7fa8b",
          "name": "test4",
          "price": 15,
        }
      }
    />
    <inject-ProductComponent-with-cart
      key="4"
      product={
        Object {
          "description": "test4",
          "id": 4,
          "image": "https://www.barbora.lt/api/Images/GetInventoryImage?id=0198db1a-2a5e-47df-94b0-562be3c7fa8b",
          "name": "test4",
          "price": 15,
        }
      }
    />
  </div>
</div>
`;

exports[`Should render product properly 1`] = `
<inject-ProductComponent-with-cart
  product={
    Object {
      "description": "test",
      "id": 1,
      "image": "https://www.barbora.lt/api/Images/GetInventoryImage?id=0198db1a-2a5e-47df-94b0-562be3c7fa8b",
      "name": "test2",
      "price": 10,
    }
  }
/>
`;
